# # startup.py
# import os
# from pathlib import Path
# import subprocess

# def download_vgg_weights():
#     url = "https://drive.google.com/uc?id=1h8emLTEflJB2dAsCUdVJNSOHgmEfmwS7"  # Full Google Drive URL
#     weights_dir = Path.home() / ".deepface" / "weights"
#     weights_dir.mkdir(parents=True, exist_ok=True)
#     weights_path = weights_dir / "vgg_face_weights.h5"

#     if weights_path.exists():
#         print("‚úÖ DeepFace weights already present.")
#         return

#     print("‚¨áÔ∏è  Downloading VGG-Face weights...")
#     try:
#         subprocess.run(
#             ["gdown", url, "-O", str(weights_path)],
#             check=True
#         )
#         print("‚úÖ Download complete.")
#     except subprocess.CalledProcessError as e:
#         print("‚ùå Failed to download DeepFace weights:", e)


def preload_facenet_model():
    from deepface import DeepFace
    
    print("üì¶ Preloading FaceNet model...")
    DeepFace.build_model("Facenet")
    print("‚úÖ FaceNet model preloaded.")

